File: herencia/Perro.cc
#include "Perro.h"
#include <iostream>
void Perro::hablar(){
  std::cout<<"Soy un perro\n";
}
void Perro::foo(){
  std::cout<<"Foo() con perro\n";
} 
void Perro::saludar(int a){
  std::cout<<"Saludar con la pata y mi edad es "<<a*2<<"\n";
}

File: JAVApOLIMORFISMO/Perro.java

package com.mycompany.polimorfismo;


public class Perro extends Animal implements Saludo{
    public Perro(String name){
        super(name);
    }
    public String toString(){
        return "Soy un perro";
    }
    public void saludar(String part){
        System.out.println("Saludar con la "+part + " y tengo la edad de 3 años");
    }
}
